truncate.cc

```diff
@@ -0,0 +1,78 @@
+#include <cstdlib>
+
+#include <algorithm>
+#include <fstream>
+#include <iostream>
+#include <iomanip>
+#include <iterator>
+#include <random>
+#include <sstream>
+#include <string>
+#include <vector>
+#include <cctype>
+#define a 0.8
+struct Point {
+  double x,y,r;
+  //std::vector<double> y;
+};
+
+int main(int argc, char** argv) {
+  std::string inputfile, outputfile;
+  if (argc == 3) {
+    inputfile = argv[1];
+    outputfile = argv[2];
+  } else {
+    std::cerr << "Incorret number of arguments." << std::endl;
+    std::exit(EXIT_FAILURE);
+  }
+  std::ifstream infile(inputfile);
+  std::ofstream ofs(outputfile);
+  std::vector<Point> points;
+  std::string line;
+ 
+  std::vector<std::string>vec;
+  if (infile.is_open()) {
+    while (std::getline(infile, line))
+      if(line!=""){
+      
+        std::string linesub = line.substr (0,50);
```

Comment:
Do you mean the substr? I don't want the each string in the vector to be too long because the I only want the second, third and fourth columns of the txt so I use the substring of the entire row. I don't know if it is reasonable.. Sorry about that.